apps/music/js/utils.js

```diff
@@ -186,3 +190,22 @@ function createListElement(option, data, index, highlight) {
 
   return li;
 }
+
+function showImage(image, url, option) {
+  // We could probably have many animation options.
+  // Add them here in the future.
+  var isFadeIn = (option === 'fadeIn');
+  image.classList.toggle('fadeIn', !isFadeIn);
```

Comment:
For 1. yes that's a mistake...I will address it.
For 2. I think I should append the img elements right after we create them, then only do `showImage` in the `AlbumArt.getCoverURL.then()` so that we can make sure the img elements are all the DOM tree, waiting for the load event to be fade-in.